### DISCUSSION TITLE
Zod inferred object doesn't work as input

### DISCUSSION THREAD

**Discussion by RichardPergament** [encoredev/encore#2353] [Suggestions]
I'm getting "error: qualified name not yet supported" when attempting to define an input object via Zod. See example below.

```
const InputZod = z.object({
    name: z.string().min(1).max(100),
});

export type Input = z.infer<typeof InputZod>;

export interface Output {}

export const post = api(
    {
        expose: true,
        auth: true,
        method: 'POST',
        path: '/some-url',
    },
    async (input: Input): Promise<Output> => { api code... }
);
```

Not entirely sure what the error message means. Is this approach not supported yet or is it a bug?

Would be cool if it were supported to reduce the amount of boilerplate (repeated model definitions), but I can understand that if that is easier said than done :-)

**Comment by ebalo55** [comment]
the issue is still there
